pkg/scale/bitvec.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale
+
+const (
+	// maxLen equivalent of `ARCH32BIT_BITSLICE_MAX_BITS` in parity-scale-codec
+	maxLen = 268435455
+	// byteSize is the number of bits in a byte
+	byteSize = 8
+)
+
+// BitVec is the implementation of the bit vector
+type BitVec struct {
```

Comment:
I strongly disagree that this type should be in the scale package.  Looks like this type is required, but we should move it to `types` package and let it implement `MarshalSCALE` and `UnmarshalSCALE` for that specific type rather than add the changes to inherently support it in the scale package.